File: NPVCalculator/Models/NPVQuery.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace NPVCalculator.Models
{
    public class NPVQuery
    {
        public int Id { get; set; }
        public decimal InitialInvestment { get; set; }
        public decimal LowerBoundRate { get; set; }
        public decimal UpperBoundRate { get; set; }
        public decimal RateIncrement { get; set; }
        public virtual ICollection<CashFlow> CashFlows { get; set; }
        public virtual ICollection<QueryResult> QueryResults { get; set; }
    }
}

File: NPVCalculator/ClientApp/src/app/classes/npvquery.ts
import { QueryResult } from './query-result';
import { CashFlow } from './cash-flow';

export class NPVQuery {
    id = 0;
    initialInvestment = 0;
    lowerBoundRate = 0;
    upperBoundRate = 0;
    rateIncrement = 0;
    cashFlows: Array<CashFlow> = new Array<CashFlow>();
    queryResults: Array<QueryResult> = new Array<QueryResult>();
}
